istio-telemetry/kiali/templates/ingress.yaml

```diff
@@ -0,0 +1,40 @@
+{{- if .Values.kiali.ingress.enabled -}}
+apiVersion: extensions/v1beta1
```

Comment:
I'm not sure Ingress will even be enabled by default - can you switch to Gateway ?

Reply:
I think it is used nginx ingress, not the old istio ingress.

Reply:
cc @jmazzitelli to correct me

Reply:
one thing we have is to provide additional doc on wiring up with istio ingress gateway instead of having the config straight in helm:  https://istio.io/docs/tasks/telemetry/gateways/

Reply:
@linsun Yes, IIRC, that ingress was not the old istio ingress. It was to enable the k8s/nginx ingress.

Reply:
@jmazzitelli Do you know the reason why gateway integration was removed for kiali before? Thanks.

Reply:
@richardwxn The gateway configs were removed two months ago from all add-ons (not just kiali) via PR https://github.com/istio/istio/pull/12309 - read that PR for the background information as to why this was done.

Reply:
I think the proper way to handle this is to add the Gateway - but not in Kiali ( since 1.1 we do not recommend Gateway or Secrets to be in the app namespace, but in the namespace of the ingress gateway ). 

In the ingress gateway, based on the flags there user should enable the gateaway and redirect to any service they need - kiali, prometheus, etc.

Ideally - the settings in gateway will also include TLS certs and auth policy requirements ( JWT tokens, client cert if we want, etc).

Reply:
In other words - ownership of the domain, certificate and external forwarding rules is in the ingress manager. 
We should recommend for 1.2 to restrict creation of Gateway with RBAC.